BEIMA.Client/src/pages/Devices/AddDevicePage.js

```diff
@@ -62,83 +72,151 @@ const AddDevicePage = () => {
     let newErrors = {};
-
+    
+    //loops through visible fields on the form
     for(let i = 0; i < formFields.length; i++){
       let formName = formFields[i].name;
       let fieldNames = Object.keys(deviceFields);
       
+      //lat lon validation
+      if (formName === 'Latitude' || formName === 'Longitude') {
+        const coordMax = formName === 'Latitude' ? MAX_LATITUDE : MAX_LONGITUDE;
+        if(!(isFinite(deviceFields[formName]) && Math.abs(deviceFields[formName]) <= coordMax)) {
+          newErrors[formName] = `${formName} value is invalid. Must be a decimal between -${coordMax} and ${coordMax}.`;
+        }
+      }
+      
+      //construct the current JSON element to be added to the insert
       if(fieldNames.includes(formName)){
-        let formJSON =  {[formName] : formFields[i].value};
+        let formJSON = '';
+        let jsonKey = convertToDbFriendly(formName);
         
-        //lat lon validation
-        if (formName === 'Latitude' || formName === 'Longitude') {
-          const coordMax = formName === 'Latitude' ? 90 : 180;
-          if(!(isFinite(formFields[i].value) && Math.abs(formFields[i].value) <= coordMax)) {
-            newErrors[formName] = `${formName} value is invalid. Must be a decimal between -${coordMax} and ${coordMax}.`;
-          }
+        //straight append when it's not part of a nested element
+        //otherwise it attaches it to the nested element
+        if(!(typeof jsonKey == 'object')){
+          formJSON =  {[jsonKey] : deviceFields[formName]};
+          Object.assign(dbJson, formJSON);
+        } else if ('fieldDbId' in jsonKey){
+          dbJson.fields[jsonKey.fieldDbId] = deviceFields[formName];
+        } else if ('location' in jsonKey){
+          dbJson.location[jsonKey.location.type] = deviceFields[formName];
         }
-        
-        Object.assign(fieldValues, formJSON);
       } else if (formName === "Device Image"){
         setDeviceImage(formFields[i].files[0]);
-        formFields[i].value = "";
       } else if (formName === "Additional Documents"){
         setAdditionalDocs(formFields[i].files);
-        formFields[i].value = "";
       }
     }
-
-    setDeviceFields(mandatoryDeviceFields);
-    setFullDeviceJSON(fieldValues);
     
     //won't deploy to azure without these until they're used elsewhere
-    console.log(fullDeviceJSON);
     console.log(deviceImage);
     console.log(deviceAdditionalDocs);
-
+    
+    //display errors when present or attempt insert when valid data is present
     if ( Object.keys(newErrors).length > 0 ) {
       setErrors(newErrors);
     } else {
-      setErrors({});
-      setSelectedDeviceType('Select Device Type');
-      for(let i = 0; i < formFields.length; i++){
-        formFields[i].value = "";
-      }
+      /*TODO Need ability to get a building ID, probably a DD on the page*/
+
+      dbJson.deviceTypeId = selectedDeviceType._id;
       
+      AddDevice(dbJson).then(response => {
```

Comment:
Given that the name of this function is 'createJson', I think we should split this part out into it's own function and just return the JSON object. Then in the add button onClick stuff we can call this chunk of code immediately after the createJson has created the json for us to pass in. Or we could just rename this function to be more descriptive about what it's doing, but I like the idea of splitting up the code a bit so it's not all in one place.

Reply:
- [ ] split json creation/submit action/change function name TBD

Reply:
createJson function only creates the save data JSON
saveDeviceToDb now calls the createJson function and submits it to the DB